src/components/dropdown-menu/DropdownMenuItemBase.tsx

```diff
@@ -85,10 +84,34 @@ export function DropdownMenuItemBase({
       </StyledLeftHandSide>
 
       <StyledSuffix>{suffix}</StyledSuffix>
+    </>
+  );
+
+  const itemContainer =
+    linkProps !== undefined ? (
+      <StyledLinkBase hasExternalIcon={false} hasNavLink {...linkProps}>
+        {itemContents}
+      </StyledLinkBase>
+    ) : (
+      itemContents
+    );
+
+  return (
+    <StyledRadixDropdownMenuItem
+      aria-label={ariaLabel}
+      {...(linkProps !== undefined ? { asChild: true } : {})}
```

Comment:
Works fine.

I'm nitpicking a bit, but the only thing I don't like is that now we have link item logic leaking into the base and every other item components.

Based on the same principle, maybe we can instead expose some internal props named `asChildWrapper` and `asChildWrapperProps` prop instead? It would behave the same way as you did, setting up the `asChild` as true. It would keep things generic in the base, and only the items that need that can be used, no need to change the other items.

To have the internal props, you might need to split `DropdownMenuItemBaseProps` in two: one that's exported with the public props and one that's not exported and only to be used by our lib. It would contain the' asChildWrapper, ' for example, or any other prop that should be public.